reactive/kubernetes_worker.py

```diff
@@ -452,14 +463,28 @@ def watch_for_changes():
         set_state('kubernetes-worker.restart-needed')
 
 
+@when('kubernetes-worker.cloud.pending',
+          'kubernetes-worker.cloud.blocked')
+@when_not('kubernetes-worker.snaps.installed', 'kube-api-endpoint.available',
+      'tls_client.ca.saved', 'tls_client.certs.saved',
+      'kube-control.dns.available', 'kube-control.auth.available',
+      'cni.available', 'kubernetes-worker.restart-needed',
+      'worker.auth.bootstrapped', 'endpoint.container-runtime.available')
+def no_container_runtime_connected():
+    hookenv.status_set(
+        'waiting',
```

Comment:
Shouldn't this be `'blocked'`?  Though it might be worth checking `goal_state` and choosing appropriately based on that.